scripts/percona-release.sh

```diff
@@ -107,6 +111,43 @@ function check_specified_repo {
   fi
 }
 #
+function check_os_support {
+   REPO_NAME=$1
+   if [[ ${PKGTOOL} = yum ]]; then
+    if [ -f /etc/os-release ]; then
+      OS_VER=$(grep VERSION_ID= /etc/os-release | awk -F'"' '{print $2}' )
+    else
+      OS_VER=$(cat /etc/system-release | awk '{print $3}' | awk -F'.' '{print $1}')
+    fi
+    reply=$(curl -Is http://repo.percona.com/${REPO_NAME}/yum/release/${OS_VER}/ | head -n 1 | awk '{print $2}')
+  elif [[ ${PKGTOOL} = "apt-get" ]]; then
+    OS_VER=$(lsb_release -sc)
+    reply=$(curl -Is http://repo.percona.com/${REPO_NAME}/apt/dists/${OS_VER}/ | head -n 1 | awk '{print $2}')
+  fi
+  if [[ ${reply} != 200 ]]; then
+      echo "Specified repository is not supported for current operation system!"
+      exit 2
+  fi
+}
+#
+function check_repo_availability {
```

Comment:
It's better to skip "check_os_support" invocation in case the provided repo name doesn't exist, in order not to confuse the end user with `"Specified repository is not supported for current operation system!"` message.

Please consider adding changes like below:
```
--- /usr/bin/percona-release	2020-07-15 22:29:06.756311266 +0000
+++ ./percona-release	2020-07-15 22:18:34.930004004 +0000
@@ -138,14 +138,19 @@
   else
     REPO_NAME=${2}
   fi
+  [[ ${REPO_NAME} == "all" ]] && return 0
   REPO_NAME=$(echo ${REPO_NAME} | sed 's/-//' | sed 's/\([0-9]\)/-\1/')
-  reply=$(curl -Is http://repo.percona.com/${REPO_NAME}/ | head -n 1 | awk '{print $2}')
+  REPO_LINK="http://repo.percona.com/${REPO_NAME}/"
+  reply=$(curl -Is ${REPO_LINK} | head -n 1 | awk '{print $2}')
   if [[ ${reply} == 200 ]]; then
     if [[ ${REPOSITORIES} != "*${REPONAME}*" ]]; then
       REPO_ALIAS=$(echo ${REPO_NAME} | sed 's/-//')
       ALIASES="${REPOSITORIES} ${REPO_ALIAS}"
       REPOSITORIES="${REPOSITORIES} ${REPO_NAME}"
     fi
+  else
+    echo "Specified repository does not exist: ${REPO_LINK}"
+    exit 2
   fi
   check_os_support ${REPO_NAME}
 }
```

The above snippet also solves the issue with inability to disable all repos and with removing "percona-release" package:
```
vagrant@ubuntu-bionic:~/ps/gnu$ sudo percona-release disable all
Specified repository is not supported for current operation system!

vagrant@ubuntu-bionic:~/ps/gnu$ sudo apt-get remove percona-release -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages will be REMOVED:
  percona-release
0 upgraded, 0 newly installed, 1 to remove and 5 not upgraded.
After this operation, 64.5 kB disk space will be freed.
(Reading database ... 121392 files and directories currently installed.)
Removing percona-release (1.0-21.generic) ...
**Specified repository is not supported for current operation system!**
dpkg: error processing package percona-release (--remove):
 installed percona-release package pre-removal script subprocess returned error exit status 2
Errors were encountered while processing:
 percona-release
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Reply:
fixed